presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PruneUnreferencedOutputs.java

```diff
@@ -197,7 +197,11 @@ public PlanNode visitJoin(JoinNode node, RewriteContext<Set<Symbol>> context)
             PlanNode left = context.rewrite(node.getLeft(), leftInputs);
             PlanNode right = context.rewrite(node.getRight(), rightInputs);
 
-            return new JoinNode(node.getId(), node.getType(), left, right, node.getCriteria(), node.getFilter(), node.getLeftHashSymbol(), node.getRightHashSymbol());
+            List<Symbol> outputSymbols = node.getOutputSymbols().stream()
```

Comment:
`ProjectionPushDown` pushes it through Union or Exchange only. Anyway, JoinNode is not as flexible as projections (cannot do computations, left symbols must be before right symbols)